### ISSUE TITLE
Renew-crypto --new-node-certificates sometimes does not complete

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1088]
Originally reported of Google Code with ID 1032.

```
<b>What steps will reproduce the problem?</b>
1. have a big cluster
2. run gnt-cluster renew-crypto --new-node-certificates


<b>What is the expected output? What do you see instead?</b>

Sometimes, the operation does not complete. Some nodes don't get a client.pem.

This is a bit of a hunch for now, but I assume that the RenewCrypto RPC call to the backend times out for bigger clusters

```
Originally added on 2015-02-26 08:56:37 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-03-03 09:01:34 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hiya Helga, so, I'm not sure if that was the issue or not. I reduced the size of my cluster, but then I noticed I never did delete the client.pem file from my master. Deleted that and things worked.
```
Originally added on 2015-04-01 21:03:03 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Alright, thanks for letting us know. Nevertheless, we had some problems of renew-crypto crashing when one node is unreachable. We fixed that and hopefully this will make it more reliable from the next release on.
```
Originally added on 2015-04-02 12:05:09 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed with:

commit 6e8f7fe5089f44336b0e2569337e4669464510e8
Author: Helga Velroyen <helgav@google.com>
Date:   Wed Mar 4 23:14:12 2015 +0100

    Renew crypto retries for non-master nodes
    
    If renewing the SSL certificate for non-master nodes fails,
    try retring two more times. Unit tests included.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit fa27c1a2a78b23bbcd308005c3b12cf7edc6e6d5
Author: Helga Velroyen <helgav@google.com>
Date:   Wed Mar 4 22:55:27 2015 +0100

    Retries for the master's SSL cert renewal
    
    If renewing the master's client SSL certificate fails, try
    two more times before giving up. Unit test included.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit fc5a17efcc36cc8b2e3cbc6d713b01a0328e77d8
Author: Helga Velroyen <helgav@google.com>
Date:   Wed Mar 4 22:22:38 2015 +0100

    Unit tests for offline nodes
    
    Add a unit test which tests that offline nodes are skipped
    by LURenewCrypto.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 6db8bbe635014f7be1a595a7fae1b28dfadcefdb
Author: Helga Velroyen <helgav@google.com>
Date:   Wed Mar 4 22:07:09 2015 +0100

    De-duplicate testing code regarding pathutils
    
    After introducing some unit tests in the last patches,
    there is some opportunity to reduce the duplicate code
    around the 'pathutils' module.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 52f7ab91f02e4976a9f4245c6b076580a35ce3ee
Author: Helga Velroyen <helgav@google.com>
Date:   Wed Mar 4 22:01:23 2015 +0100

    Make LURenewCrypto handle unreachable nodes properly
    
    Currently an unreachable node can make LURenewCrypto fail
    completely. This patch adds a unit test for it, and
    improves the error handling of unreachable nodes in
    a way, that the rest of the nodes are still handled
    properly.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit c017cd6a8f13536da8c38d9c93d1502b7bf15db5
Author: Helga Velroyen <helgav@google.com>
Date:   Wed Mar 4 21:31:40 2015 +0100

    Error handling on failed SSL cert renewal for master
    
    When the recreation of the master's SSL client certificate
    fails, LURenewCrypto did not conclude very graciously.
    This patch adds unit tests for this case and improves
    the error handling.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 01fa4e6cfd039a40fd41cafc999dd6aafa24bb94
Author: Helga Velroyen <helgav@google.com>
Date:   Wed Mar 4 18:41:45 2015 +0100

    Unit test for LURenewCrypto's valid case
    
    This adds a unit test which tests a successful run of
    LURenewCrypto. While writing this, some options for
    improvement became apparent and are fixed in this patch
    as well.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 890b7bfb57e7545df1650e5bd4c670001f391e05
Author: Helga Velroyen <helgav@google.com>
Date:   Wed Mar 4 19:14:28 2015 +0100

    Mock support for pathutils
    
    For better LURenewCrypto unit tests, we need to be able to
    mock the pathutils module. This makes it necessary to add
    some support to the testutils for patching this module.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit f6ccfdbe2ecfc432a8a76adc781b1cbd0e88c773
Author: Helga Velroyen <helgav@google.com>
Date:   Wed Mar 4 20:46:14 2015 +0100

    Increase timeout of crypto token RPC
    
    Currently, gnt-cluster renew-crypto is not very robust. It
    fails completely as soon as one node does not create new
    crypto tokens fast enough. To give the RPC more time,
    increase the timeout.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2015-04-02 12:08:04 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:05 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:42 +0000 UTC.

Changed State: Released
